runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -275,27 +275,110 @@ void J9::RecognizedCallTransformer::process_java_lang_StringUTF16_toBytes(TR::Tr
-   prepareToReplaceNode(node);
+   TR::CFG *cfg = comp()->getFlowGraph();
 
-   int32_t byteArrayType = fej9->getNewArrayTypeFromClass(fej9->getByteArrayClass());
+   // The implementation of java.lang.StringUTF16.toBytes(char[],int,int) will
+   // throw a NegativeArraySizeException or OutOfMemoryError if the specified
+   // length is outside the range [0,0x3fffffff].  In order to avoid deciding
+   // which to throw in the IL, fallback to the out-of-line call if the length
+   // is negative or too great.  Otherwise, create the byte array and copy the
+   // input char array to it with java.lang.String.decompressedArrayCopy
+   //
+   // Before:
+   //
+   // +----------------------------------------+
+   // | treetop                                |
+   // |   acall  java/lang/StringUTF16.toBytes |
+   // |     aload  charArr                     |
+   // |     iload  off                         |
+   // |     iload  len                         |
+   // +----------------------------------------+
+   //
+   // After:
+   //
+   // ifCmpblock
+   // +----------------------------------------+
+   // | astore charArrTemp                     |
+   // |   aload  charArr                       |
+   // | istore offTemp                         |
+   // |   iload  off                           |
+   // | istore lenTemp                         |
+   // |   iload  len                           |
+   // | ifiucmpgt --> fallbackPathBlock  -----------------+
+   // |   iload lenTemp                        |          |
+   // |   iconst 0x3fffffff                    |          |
+   // +--------------------+-------------------+          |
+   //                      |                              |
+   // fallThroughPathBlock V                              |
+   // +----------------------------------------+          |
+   // | astore result                          |          |
+   // |   newarray jitNewArray                 |          |
+   // |     ishl                               |          |
+   // |       iload lenTemp                    |          |
+   // |       icosnt 1                         |          |
+   // |     iconst 8   ; array type is byte    |          |
+   // | treetop                                |          |
+   // |   call java/lang/String.decompressedArrayCopy     |
+   // |     aload charArrTemp                  |          |
+   // |     iload offTemp                      |          |
+   // |     ==>newarray                        |          |
+   // |     iconst 0                           |          |
+   // |     iload lenTemp                      |          |
+   // | goto joinBlock   ----------------------------+    |
+   // +----------------------------------------+     |    |
+   //                                                |    |
+   //                      +------------------------------+
+   //                      |                         |
+   // fallBackPathBlock    V (freq 0) (cold)         |
```

Comment:
Minor nitpick: "fallBack" or "fallback"?  See also line 315, and variable names at lines 403-404 and 420.

Reply:
Thanks!  I've fixed up usage of "fallback" versus "fall back" in commit f3d338d270e56ad6adf35e79232fb53af208d08a.

Reply:
Once changes have been approved, I will squash the commit that corrects the typos.